Goal: ???

I think it should be dropped. The MAXHOSTNAMELEN macro is not used.
It is usually used for the gethostname buffer.


Index: shadow-4.0.3/src/login.c
===================================================================
--- shadow-4.0.3.orig/src/login.c	2005-05-29 11:07:14.638975000 +0200
+++ shadow-4.0.3/src/login.c	2005-05-29 11:09:47.018975000 +0200
@@ -58,6 +58,11 @@
 #include <sys/proc.h>
 #include <sys/sysi86.h>
 #endif
+
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN 64
+#endif
+
 #ifdef RADIUS
 /*
  * Support for RADIUS authentication based on a hacked util-linux login
